src/components/editor-page/editor-pane/autocompletion/code-block.ts

```diff
@@ -5,45 +5,103 @@
  */
 
 import { Editor, Hint, Hints, Pos } from 'codemirror'
-import { findWordAtCursor, Hinter, search } from './index'
+import { findWordAtCursor, generateHintListByPrefix, Hinter } from './index'
+
+type highlightJsImport = typeof import('../../../common/hljs/hljs')
 
 const wordRegExp = /^```((\w|-|_|\+)*)$/
 let allSupportedLanguages: string[] = []
 
-const codeBlockHint = (editor: Editor): Promise<Hints | null> => {
-  return import(/* webpackChunkName: "highlight.js" */ '../../../common/hljs/hljs').then(
-    (hljs) =>
-      new Promise((resolve) => {
-        const searchTerm = findWordAtCursor(editor)
-        const searchResult = wordRegExp.exec(searchTerm.text)
-        if (searchResult === null) {
-          resolve(null)
-          return
-        }
-        const term = searchResult[1]
-        if (allSupportedLanguages.length === 0) {
-          allSupportedLanguages = hljs.default
-            .listLanguages()
-            .concat('csv', 'flow', 'html', 'js', 'markmap', 'abc', 'graphviz', 'mermaid', 'vega-lite')
-        }
-        const suggestions = search(term, allSupportedLanguages)
-        const cursor = editor.getCursor()
-        if (!suggestions) {
-          resolve(null)
-        } else {
-          resolve({
-            list: suggestions.map(
-              (suggestion: string): Hint => ({
-                text: '```' + suggestion + '\n\n```\n',
-                displayText: suggestion
-              })
-            ),
-            from: Pos(cursor.line, searchTerm.start),
-            to: Pos(cursor.line, searchTerm.end)
-          })
-        }
+/**
+ * Fetches the highlight js chunk.
+ * @return the retrieved highlight js api
+ */
+const loadHighlightJs = async (): Promise<highlightJsImport | null> => {
+  try {
+    return await import('../../../common/hljs/hljs')
+  } catch (error) {
+    console.error("can't load highlight js", error)
```

Comment:
Maybe use the new notification system instead.

Reply:
Hm. Maybe. But I'm not sure how much information we should provide. Maybe just "System error occured while doing xyz. Check browser console and report the error if it occurs again." ?
Notifications are for end users.

Reply:
Yes, it's just that end-users get notified that there's a problem and that they may seek help in our matrix chat.

Most people don't look into the browser console if something isn't working and therefore just writing to the console is like a silent failure (which is bad UX).

Reply:
done